go/porcelain/deploy.go

```diff
@@ -538,10 +538,17 @@ func (n *Netlify) uploadFile(ctx context.Context, d *models.Deploy, f *FileBundl
 			context.GetLogger(ctx).WithError(operationError).Errorf("Failed to upload file %v", f.Name)
 			apiErr, ok := operationError.(apierrors.Error)
 
-			if ok && apiErr.Code() == 401 {
-				sharedErr.mutex.Lock()
-				sharedErr.err = operationError
-				sharedErr.mutex.Unlock()
+			if ok {
+				if apiErr.Code() == 401 {
+					sharedErr.mutex.Lock()
+					sharedErr.err = operationError
+					sharedErr.mutex.Unlock()
+				}
+
+				// TODO this changes retry behaviour for the fileUpload case as well. OK?
+				if apiErr.Code()/100 == 4 {
```

Comment:
I would voice the same concern as @biruwon pointed out here. I'm concerned that retrying all the 4xx calls might be a stretch? (i.e. should we retry 429's too?)

Reply:
Updated to just skip the retry for 400 and 422 status codes 🙂